src/tests/PokemonDetails.test.js

```diff
@@ -8,43 +9,41 @@ describe('Testing structure PokemonDetails.js', () => {
   it('Testa se as informações detalhadas do Pokémon são mostradas na tela', () => {
     const pikachu = pokemons[0];
 
-    const { getByText, history, queryByRole, container } = renderWithRouter(<App />);
+    const { history } = renderWithRouter(<App />);
 
     history.push(`/pokemons/${pikachu.id}`);
 
-    const pikachuName = getByText(/Pikachu details/i);
-    const link = queryByRole('link', { name: 'More details' });
-    const h2 = container.querySelectorAll('h2');
-    const textH2 = h2[1];
-    const getAllP = document.querySelectorAll('p');
-    const summaryP = getAllP[3];
+    const pikachuName = screen.getByText(/Pikachu details/i);
+    const link = screen.queryByRole('link', { name: 'More details' });
+    const SummaryTitle = screen.getByRole('heading', { name: /Summary/i });
+    const SummaryContent = screen.getByText(/This intelligent Pokémon/i);
 
     expect(pikachuName).toBeInTheDocument();
     expect(link).not.toBeInTheDocument();
-    expect(textH2).toBeInTheDocument();
-    expect(textH2.innerHTML).toBe(' Summary ');
-    expect(summaryP.innerHTML).toBe(`${pikachu.summary}`);
+    expect(SummaryTitle).toBeInTheDocument();
+    expect(SummaryContent).toBeInTheDocument();
   });
 
   it('Testa se existe uma seção com os mapas contendo localizações do pokémon', () => {
-    const { getByText, getAllByRole, history, container } = renderWithRouter(<App />);
+    const { history } = renderWithRouter(<App />);
     const pikachu = pokemons[0];
 
     history.push(`/pokemons/${pikachu.id}`);
 
-    const h2 = container.querySelectorAll('h2');
-    const selecH2 = h2[2];
-    const images = getAllByRole('img');
+    const GameLocationsTitle = screen.getByRole('heading', {
+      name: /Game Locations of Pikachu/i,
+    });
+    const images = screen.getAllByRole('img');
     const img1 = images[1];
     const img2 = images[2];
     const two = 2;
-    const location1 = getByText(/Kanto Viridian Forest/i);
-    const location2 = getByText(/Kanto Power Plant/);
+    const location1 = screen.getByText(/Kanto Viridian Forest/i);
+    const location2 = screen.getByText(/Kanto Power Plant/);
 
     expect(location1.innerHTML).toBe('Kanto Viridian Forest');
     expect(location2.innerHTML).toBe('Kanto Power Plant');
     expect(pikachu.foundAt.length).toBe(two);
-    expect(selecH2.innerHTML).toBe(`Game Locations of ${pikachu.name}`);
+    expect(GameLocationsTitle).toBeInTheDocument();
```

Comment:
```suggestion
    expect(gameLocationsTitle).toBeInTheDocument();
```

Se mudar na linha 33, teria que mudar aqui também :)

Reply:
Ok!